server/sonar-server/src/main/java/org/sonar/server/license/ws/ListAction.java

```diff
@@ -0,0 +1,188 @@
+import org.sonarqube.ws.Licenses.ListWsResponse;
+
+import static org.sonar.api.CoreProperties.PERMANENT_SERVER_ID;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.server.ws.WsUtils.writeProtobuf;
+import static org.sonarqube.ws.client.license.LicensesWsParameters.ACTION_LIST;
+
+public class ListAction implements WsAction {
+
+  private static final String ALL_SERVERS_VALUE = "*";
+
+  private final UserSession userSession;
+  private final PropertyDefinitions definitions;
+  private final DbClient dbClient;
+
+  public ListAction(UserSession userSession, PropertyDefinitions definitions, DbClient dbClient) {
+    this.userSession = userSession;
+    this.definitions = definitions;
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    context.createAction(ACTION_LIST)
+      .setDescription("List licenses settings.<br>" +
+        "Requires Administer System' permission")
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setSince("6.1")
+      .setInternal(true)
+      .setHandler(this);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(SYSTEM_ADMIN);
+
+    DbSession dbSession = dbClient.openSession(true);
+    try {
+      writeProtobuf(doHandle(dbSession), request, response);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private ListWsResponse doHandle(DbSession dbSession) {
+    Set<String> settingsKeys = definitions.getAll().stream()
+      .filter(definition -> definition.type().equals(PropertyType.LICENSE))
+      .map(PropertyDefinition::key)
+      .collect(Collectors.toSet());
+    settingsKeys.add(PERMANENT_SERVER_ID);
+    List<PropertyDto> properties = dbClient.propertiesDao().selectGlobalPropertiesByKeys(dbSession, settingsKeys);
+    return new ListResponseBuilder(settingsKeys, properties).build();
+  }
+
+  private static class ListResponseBuilder {
+    private final Optional<String> serverId;
+    private final Map<String, PropertyDto> licenseSettingsByKey;
+    private final Set<String> licenseSettingsKeys;
+
+    ListResponseBuilder(Set<String> licenseSettingsKeys, List<PropertyDto> properties) {
+      this.serverId = getServerId(properties);
+      this.licenseSettingsKeys = licenseSettingsKeys.stream()
+        .filter(key -> !key.equals(PERMANENT_SERVER_ID))
+        .collect(Collectors.toSet());
+      this.licenseSettingsByKey = properties.stream().collect(Collectors.toMap(PropertyDto::getKey, Function.identity()));
+    }
+
+    ListWsResponse build() {
+      ListWsResponse.Builder wsResponse = ListWsResponse.newBuilder();
+      licenseSettingsKeys.forEach(key -> wsResponse.addLicenses(buildLicense(key, licenseSettingsByKey.get(key))));
+      return wsResponse.build();
+    }
+
+    private Licenses.License buildLicense(String key, @Nullable PropertyDto setting) {
```

Comment:
should we rename PropertyDto to SettingDto? I'd like our code to be more consistent than it is today